### PR TITLE
Plateau from string

### PR DESCRIPTION
**Pull request by WSeegers**
Implemented a string parser for Plateau struct #9

### CODE REVIEW

File: src/models/plateau/parser.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		let width = match map.find(|c| c == '\n') {
+			Some(w) => w,
+			none => map.len(),
+		};
+
+		let mut height = 0;
+		let mut cells = Vec::new();
+		for (y, row) in map.lines().enumerate() {
+			if row.len() != width {
+				return Err(format!("Rows sizes are inconsistant"));
```
Comment (GJSteyn) [inline src/models/plateau/parser.rs]: Doing a simple "String::from()" here might be better if the string won't be formatted

File: src/models/plateau/parser.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+impl TryFrom<String> for Plateau {
+	type Error = String;
+
+	fn try_from(map: String) -> Result<Self, String> {
+		if map.is_empty() {
+			return Err(String::from("Map is empty"));
+		}
+		let width = match map.find(|c| c == '\n') {
+			Some(w) => w,
+			none => map.len(),
```
Comment (GJSteyn) [inline src/models/plateau/parser.rs]: Pretty sure this should be "None" - "none" would match anything in the same way that "_" does

Comment (GJSteyn) [reply src/models/plateau/parser.rs]: So it should have almost the same effect, except that the value "None" should get captured by "none"

File: src/models/plateau/parser.rs (hunk 1 above)
Comment (GJSteyn) [reply src/models/plateau/parser.rs]: Also "inconsistent" :P

File: src/models/plateau/parser.rs (hunk 2 above)
Comment (WSeegers) [reply src/models/plateau/parser.rs]: Fixed

File: src/models/plateau/parser.rs (hunk 1 above)
Comment (WSeegers) [reply src/models/plateau/parser.rs]: Fixed

**Pull request merged**